### ISSUE TITLE
[feature request] mark read/unread based on age

### ISSUE THREAD

**Issue by pcause** [martinrotter/rssguard#248]
It would be very useful to have a capability to mark all items in a feed/category/account read/unread based on being older (for mark read) or newer (for mark unread) hours/days/weeks.

a pop up to get the specific count and type (hours/days/weeks) could be used.  allow the user to set a default or better just use the last value entered as a default.

**Comment by martinrotter** [comment]
> a pop up to get the specific count and type (hours/days/weeks) could be used

@pcause Could you elaborate, what exactyl "a pop up to get the specific **count**" means?

**Comment by pcause** [comment]
You could say

Mark items read [x]/ unread [x]
Older/newer than [nn] [hours/days/weeks/months]

user read/undread are checkboxes
older than [for read] / new than [for unread] and then the number of hours/days/weeks/months

So, for example, lets [ick an interval of 12 hours.  For mark read items,  marks all items that are unread and received/updated more than 12 hours ago read.  For marking unread, you would mark any items that are read as unread of they were recieved/updated less than 12 hours ago.  Think it is a pretty straight forward SQL statement. something like

update table set read_state - new value where feed is XX and read_state is not_new_value  and ( received > interval or updated > interval)

but I am not a sql person

**Comment by nullrend** [comment]
For the UI I think the best location would be an icon on the Messages List toolbar with options for "Older than one day" and "Older than one week". Perhaps an option in the settings to define custom timetables? Having these two cover most of the use cases, I think.

**Comment by pcause** [comment]
yes, although if you had something to set a value (in days) and then a single to execute subsequnet use.  so button to set interval (default can be one day) and then a second to mark read/unread where mouse over shows interval.

**Comment by martinrotter** [comment]
This can be now simulated via relatively new article list filter options which allow to filter out articles based on several conditions, remaining ones then can be marked read unread.

I do not plan implementing this by myself, all PRs are welcomed.
